### ISSUE TITLE
io.popen() that supports getting return code

### ISSUE THREAD

**Issue by knazarov** [tarantool/tarantool#3769]
Currently the usage of io.popen() looks like this:

```lua
local file = io.popen("<shell command>")
local result = file:read("*all")
local retcode = result:close()
```

In lua 5.2, `retcode` will contain the return code of shell command. However, in lua 5.1 and luajit it looks to always be `true`. This makes it impossible to get the return value of the shell command.

I suggest to make `result:close()` more in line with lua 5.2, or add something like `fio.popen()`

**Comment by Totktonada** [comment]
See #4031.

**Comment by igormunkin** [comment]
Lua 5.2 is not backward compatible with Lua 5.1 ergo Tarantool can't be simply built with `LUAJIT_ENABLE_LUA52COMPAT`. As @Totktonada mentioned, `popen.popen` is implemented in scope of #4031, feel free to use it for your purposes.

Closing the issue with wontfix reason.
